deploy/ray-operator/controllers/raycluster_controller.go

```diff
@@ -179,33 +180,59 @@ func (r *RayClusterReconciler) Reconcile(request reconcile.Request) (reconcile.R
 		}
 	}
 
+	// update cluster when idList generated by controller
+    _ = r.Update(context.Background(), instance)
+
 	return reconcile.Result{}, nil
 }
 
 // Build cluster instance pods.
 func (r *RayClusterReconciler) buildPods(instance *rayiov1alpha1.RayCluster) []corev1.Pod {
 	var pods []corev1.Pod
+	var extensions []rayiov1alpha1.Extension
 	if instance.Spec.Extensions != nil && len(instance.Spec.Extensions) > 0 {
 		for _, extension := range instance.Spec.Extensions {
-			var i int32 = 0
-			for i = 0; i < *extension.Replicas; i++ {
-				podType := fmt.Sprintf("%v", extension.Type)
-				podName := instance.Name + common.DashSymbol + extension.GroupName + common.DashSymbol + podType + common.DashSymbol + utils.FormatInt32(i)
-				podConf := common.DefaultPodConfig(instance, podType, podName)
-				podConf.Extension = extension
-				pod := common.BuildPod(podConf)
-				// Set raycluster instance as the owner and controller
-				if err := controllerutil.SetControllerReference(instance, pod, r.Scheme); err != nil {
-					log.Error(err, "Failed to set controller reference for raycluster pod")
-				}
-				pods = append(pods, *pod)
-			}
-		}
-	} else {
-		log.Info("RayCluster extensions are nil or empty")
-	}
+            if extension.IdList != nil && len(extension.IdList) >0 {
+                for _, podName := range extension.IdList {
+                    podType := fmt.Sprintf("%v", extension.Type)
+                    pods = r.createAndBindPod(instance, podType, podName, extension, pods)
+                }
+            } else {
+                var i int32 = 0
+                for i = 0; i < *extension.Replicas; i++ {
+                    podType := fmt.Sprintf("%v", extension.Type)
+                    podName := ""
+                    if extension.GroupName == "" {
+                        podName = instance.Name + common.DashSymbol + extension.GroupName + common.DashSymbol + podType + common.DashSymbol + uuid.New()[0:6]
+                    } else {
+                        podName = instance.Name + common.DashSymbol + podType + common.DashSymbol + uuid.New()[0:6]
```

Comment:
Use a utility function to shorten text concat

Reply:
Extract the method to generate the podName at the latest version.